Makefile

```diff
@@ -5,7 +5,14 @@ GOOS     = $(shell go env GOOS)
 GOARCH   = $(shell go env GOARCH)
 GOBIN    = $(GOPATH)/bin/$(GOOS)-$(GOARCH)
 
-GO_BUILD_FLAGS=-v
+GO_BUILD_FLAGS=-v -ldflags="-s -w"
+
+# Allow users to pass in OS build options
+ifeq ($(TARGET),rhel7)
+	OS := rhel7
+else
+	OS := alpine
+endif
```

Comment:
This variable name should be more distinctive, and the same throughout. Maybe 'BASE_OS'.

Instead of an if here, just provide a default: `BASE_OS ?= alpine`  Then `make BASE_OS=rhel7 ...` works as expected.